src/Stencil.php

```diff
@@ -0,0 +1,294 @@
+ */
+
+use Glpi\Application\View\TemplateRenderer;
+use Glpi\Features\ZonableModelPicture;
+
+class Stencil extends CommonDBChild implements ZonableModelPicture
+{
+    public static $itemtype = "itemtype";
+    public static $items_id = "items_id";
+
+    private CommonGLPI $item;
+
+    public static function getTypeName($nb = 0): string
+    {
+        return __('Stencil');
+    }
+
+    public static function getIcon(): string
+    {
+        return 'ti ti-shape';
+    }
+
+    public static function getTable($classname = null): string
+    {
+        return 'glpi_stencils';
+    }
+
+    public function canViewStencil(): bool
+    {
+        return $this->getStencilItem()->canView();
+    }
+
+    public function canUpdateStencil(): bool
+    {
+        return $this->getStencilItem()->canUpdate();
+    }
+
+    public function canCreateStencil(): bool
+    {
+        return $this->getStencilItem()->canCreate();
+    }
+
+    public function canPurgeStencil(): bool
+    {
+        return $this->getStencilItem()->canPurge();
+    }
+
+    public static function getStencilFromItem(CommonGLPI $item): Stencil
+    {
+        $itemtype = $item::getType();
+        $items_id = $item->getID();
+
+        switch ($item->getType()) {
+            case NetworkEquipment::getType():
+            case NetworkEquipmentModel::getType():
+                $stencil = new NetworkEquipmentModelStencil();
+                break;
+        }
+
+        if (isset($stencil)) {
+            $stencil->getFromDBByCrit([
+                'itemtype' => $itemtype,
+                'items_id' => $items_id,
+            ]);
+            $stencil->item = $item;
+
+            return $stencil;
+        }
+
+        return null;
+    }
+
+    public static function getStencilFromID(int $id): Stencil
+    {
+        global $DB;
+
+        $iterator = $DB->request([
+            'SELECT' => [
+                'glpi_stencils.id',
+                'glpi_stencils.itemtype',
+                'glpi_stencils.items_id',
+            ],
+            'FROM'   => self::getTable(),
+            'WHERE'  => [
+                'glpi_stencils.id' => $id,
+            ],
+        ]);
+
+        if ($iterator->count() == 1) {
+            $itemtype = $iterator->current()['itemtype'];
+            $items_id = $iterator->current()['items_id'];
+
+            $item = getItemForItemtype($itemtype);
+            if ($item->getFromDB($items_id)) {
+                return self::getStencilFromItem($item);
+            }
+        }
+
+        return null;
+    }
+
+    public function getStencilItem(): ?CommonGLPI
+    {
+        if (isset($this->item)) {
+            return $this->item;
+        }
+
+        if (!isset($this->fields['itemtype']) || !isset($this->fields['items_id'])) {
+            return null;
+        }
+
+        $itemtype = $this->fields['itemtype'];
+        $items_id = $this->fields['items_id'];
+
+        $item = getItemForItemtype($itemtype);
```

Comment:
A bit of an edge case but `getItemForItemtype` can return `false`, which mean `$item` will be false and we will get an exception when calling `getFromDB()` on it.

Granted that this is not very likely to happen ;)